www/js/lib/uiUtil.js

```diff
@@ -32,6 +32,40 @@ if (webpMachine) {
 }
 
 define(rqDef, function(settingsStore) {
+
+    /**
+     * Displays a Bootstrap alert or confirm dialog box depending on the options provided
+     * 
+     * @param {String} label The modal's label or title which appears in the header 
+     * @param {String} message The alert message to display in the body of the modal 
+     * @param {Boolean} isConfirm If true, the modal will be a confirm dialog box, otherwise it will be an alert 
+     * @param {String} declineButtonText The text to display on the decline button (optional, Default = "Cancel") 
+     * @param {String} approveButtonText  The text to display on the approve button (optional, Default = "Confirm")
+     * @returns {Promise<Boolean>} A promise which resolves to true if the user clicked Confirm, false if the user clicked Cancel, backdrop or the cross(x) button
+     */
+    function systemAlert(label, message, isConfirm, declineButtonText, approveButtonText) {
+        declineButtonText = declineButtonText || "Cancel";
```

Comment:
To simplify future translations of the UI, it would be better to avoid setting these labels here.
The already are in the HTML: it would be better to test if `declineButtonText` is set, and to change `document.getElementById("declineModal")` only if it is.
(same for approve, of course)
NB: in the HTML, the label is "Close" instead of "Cancel": "Cancel looks better to me"

Reply:
@mossroy  One doubt here, if someone passes, say, `Lorem` as approve button's text in some `systemAlert()` call. Then, this code will set the `document.getElementById("approveModal").textContent` to `Lorem`. So, in any next modal call at some different place in the main code where `approveButtonText` is not passed, what will it show (`Lorem` or `Confirm`)? Is the value set by `textContent` API persistent?

This might be silly, but a bit confused here...

Reply:
You're right, this won't work. It would show `Lorem`.
A solution would be to save the original value at the beginning of this function and restore it at the end.

But, in fact, it seems to me that these parameters are currently never passed by the existing calls.
If you wish, you might remove them: we'll add them again later if we need

Reply:
I would prefer to keep this feature around. Will fix by restoring the default value at the end. But, what's the issue with the current implementation?
> To simplify future translations of the UI

Didn't get this point earlier..

Reply:
Translation of the UI is a separate issue: #13 

But, to be able to do that in the future, we will need a clean separation between the text of the UI and the app logic.
All the text should be in the HTML files (I know it's already not always the case): we should avoid having the "Cancel" and "Confirm" strings in the javascript code

Reply:
@mossroy Restoring these button labels, in the end, isn't this same as setting button labels from the javascript. Like, in the earlier implementation, `declineButtonText = declineButtonText || "Cancel";`. We're now just doing this in 2 steps. I didn't get what will change here?

Reply:
There are many javascript translation frameworks that are able to replace the HTML strings by translated one (in the language chosen by the user). Some do that with different HTML files, other ones do that by parsing the HTML and replace the string in it etc.
But if the string "Cancel" is in a javascript file, it will be harder to translate it, because the framework won't be able to guess this is a string that needs to be translated.
Anyway, don't bother too much on that. If you're not comfortable with that, leave the earlier implementation, and we'll work on it later

Reply:
@gaurav7019 If you decide to do that as part of this PR, then you can easily grab the text from the HTML by doing something like `document.getElementById('declineModal').innerText`. Once assigned to variable, that can replace the `'Cancel'` in `declineButtonText = declineButtonText || 'Cancel';`

Reply:
Yes, will do this.